File: JavaScriptObjCTest/JavaScriptObjCTest/CustomJSObject.m
//
//  CustomJSObject.m
//  JavaScriptObjCTest
//
//  Created by Antoine Quint on 20/06/14.
//  Copyright (c) 2014 Studio Luco. All rights reserved.
//

#import "CustomJSObject.h"

@implementation CustomJSObject

@synthesize testProperty;

- (NSString *)testProperty
{
    NSLog(@"get testProperty");
    return @"testProperty";
}

- (void)setTestProperty:(NSString *)testProperty
{
    NSLog(@"set testProperty");
}

- (NSString *)testFunction
{
    NSLog(@"called testFunction");
    return @"testFunction()";
}

@end

File: JavaScriptSwiftTest/JavaScriptSwiftTest/CustomJSObject.swift
//
//  CustomJSObject.swift
//  JavaScriptSwiftTest
//
//  Created by Antoine Quint on 20/06/14.
//  Copyright (c) 2014 Studio Luco. All rights reserved.
//

import Foundation
import JavaScriptCore

protocol JSCustomObjectExport : JSExport {
    var testProperty : String { get set }
    func testFunction() -> String
}

class JSCustomObject : NSObject, JSCustomObjectExport {
    var testProperty : String {
    get {
        println("get testProperty")
        return "testProperty"
    }
    set(newProperty) {
        println("set testProperty")
    }
    }
    
    func testFunction() -> String {
        println("called testFunction")
        return "testFunction()"
    }
}
